its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/SolutionKindTest.java

```diff
@@ -104,27 +101,27 @@ void testCSharpAllFlat() throws IOException {
     context.build.addArgument("CSharpAllFlat.sln");
     context.runAnalysis();
 
-    assertThat(getComponent(context.projectKey + ":Common.cs")).isNotNull();
+    assertThat(TestUtils.getComponent(context.projectKey + ":Common.cs")).isNotNull();
```

Comment:
Just to make sure: What happens when the component can not be found? Do we get null here? Maybe it is better to assert more than just not null but also `language`, `path`, and/or `qualifier` as well.

Also for the other asserts below.

Reply:
Current implementation as it is now in this PR will throw a useless exception with 404. Finding the correct component ID is a difficult guessing game.

That's why #2437 removes `getComponent` and replaces it with reasonable assertions on `listComponents` collection. Where the error message displayes actual/expected/missing/unexpected values => easy to understand.